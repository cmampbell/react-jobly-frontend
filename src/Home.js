import React from 'react'
import { Link, useOutletContext } from 'react-router-dom';
import './Home.css'

const Home = () => {
    const { currentUser } = useOutletContext()

    return (
        currentUser ?
            <>
                <h2>Welcome back {currentUser}</h2>
                <Link to="/jobs">Look at jobs</Link>
                <Link to="/companies">Look at companies</Link>
            </>
            :
            <>
                <h2>You must be a user to see jobs and companies</h2>
                <Link to="/login">Login</Link>
                <Link to="/signup">Signup</Link>
            </>
    )
}

export default Home;