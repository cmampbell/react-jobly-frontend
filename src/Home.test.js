import React from 'react'
import { render } from '@testing-library/react';
import Home from './Home'
import { MemoryRouter } from 'react-router-dom';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useOutletContext: function () {
        return {
            currentUser: 'test'
        }
    }
})
);

describe('Smoke and snapshot tests for Home.js', () => {
    it('should render without crashing', () => {
        render(<Home />, {wrapper: MemoryRouter})
    })

    it('should match snapshot', async () => {
        const { asFragment, findByText } = render(<Home />, {wrapper:MemoryRouter})
        expect(asFragment()).toMatchSnapshot();
        expect(await findByText('Welcome back test')).toBeInTheDocument()
      })
})