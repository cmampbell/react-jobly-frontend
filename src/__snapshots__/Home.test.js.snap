// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot tests for Home.js should match snapshot 1`] = `
<DocumentFragment>
  <h2>
    Welcome back test
  </h2>
  <a
    href="/jobs"
  >
    Look at jobs
  </a>
  <a
    href="/companies"
  >
    Look at companies
  </a>
</DocumentFragment>
`;
